src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler.properties

```diff
@@ -234,8 +234,20 @@ compiler.err.switch.expression.no.result.expressions=\
     switch expression does not have any result expressions
 
 # 0: name
-compiler.err.call.must.be.first.stmt.in.ctor=\
-    call to {0} must be first statement in constructor
+compiler.err.call.must.only.appear.in.ctor=\
+    calls to {0}() may only appear within constructors
```

Comment:
Note that super()/this() in the JLS lingo is called "explicit constructor invocation".

Reply:
@GavinBierman do you think the compiler message should mention the official JLS terminology?

Reply:
For instance:

```
explicit constructor invocation may only appear within constructors
```

```
redundant explicit constructor invocation
```

```
explicit constructor invocation not allowed here
```

```
'return'' not allowed before explicit constructor invocation
```

Reply:
@mcimadamore Yes, I like the first. Something like "explicit constructor invocation may only appear within a constructor body" is good.

(The JLS actually says that an (explicit) constructor invocation, i.e. this(...) or super(...), is not a statement, but just a thing that can appear in a constructor body.)

Reply:
Updated error messages in 738c10a57de.

Reply:
Good job! I like the new messages.